contracts/Polygon_SpokePool.sol

```diff
@@ -164,6 +164,30 @@ contract Polygon_SpokePool is IFxMessageProcessor, SpokePool {
         _wrap();
     }
 
+    /**
+     * @notice These functions can send an L2 to L1 message so we are extra cautious about preventing a griefing vector
+     * whereby someone batches this call with a bunch of other calls and produces a very large L2 burn transaction.
+     * This might make the L2 -> L1 message fail due to exceeding the L1 calldata limit.
+     */
+
+    function executeUSSRelayerRefundLeaf(
+        uint32 rootBundleId,
+        USSRelayerRefundLeaf memory relayerRefundLeaf,
+        bytes32[] memory proof
+    ) public override {
+        if (relayerRefundLeaf.amountToReturn > 0 && AddressLibUpgradeable.isContract(msg.sender)) revert NotEOA();
```

Comment:
I don't think this is an adequate check. OZ removed this because of how easy it is to misuse.

The issue is that this is checking the code size of the calling address, but if called inside a constructor, the code size is 0, but it can talk to other contracts atomically (which is what we're trying to avoid).

I think there are only two ways to avoid this (we've had this problem previously):
1. Force the msg.sender to pass in a signature, since it's impossible for a contract to generate a cryptographic signature.
2. Verify that tx.origin (if that's supported on polygon) equals msg.sender, since a contract can't be the originator of a transaction.

Reply:
Good call, done and added tests specifically for this constructor vector